### PR TITLE
Implement Fetching Course Grades

### PR DESCRIPTION
**Pull request by ecoist-ste**
- Added one more case to CanvasRequest to accommodate fetching enrollments.
- Used CanvasService to fetch enrollments

### CODE REVIEW

**Comment by azooz2003-bit** [comment]
Looking great @ecoist-ste ! Could we address [this](https://github.com/gtiosclub/CanvasPlusPlayground/pull/7#discussion_r1759930517)? Or is it not possible?

**Comment by azooz2003-bit** [comment]
Also, is the other Grades PR no longer relevant?

File: CanvasPlusPlayground.xcodeproj/project.pbxproj
Comment (rahulon12) [inline CanvasPlusPlayground.xcodeproj/project.pbxproj:1]: It seems like there are changes in this file that are more than what is indicated by the changed files themselves. I think it's just a reordering but maybe we should make sure this doesn't get messed up. @azooz2003-bit any thoughts?

File: CanvasPlusPlayground/Views/CourseGradeView.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            } else {
+                Text("Final Grade unavailable")
+            }
+            
+        }
+        .task {
+            await gradeManager.fetchEnrollments()
+            
+            for enroll in self.gradeManager.enrollments {
+                if let idLHS = enroll.courseID, let idRHS = self.course.id {
```
Comment (rahulon12) [inline CanvasPlusPlayground/Views/CourseGradeView.swift:52]: I agree with Aziz on moving this to the view model. You could also combine this into one if statement:

```suggestion
                if let idLHS = enroll.courseID, 
                   let idRHS = self.course.id, 
                   idLHS == idRHS {
```

**Comment by ecoist-ste** [comment]
> Also, is the other Grades PR no longer relevant?

Yes. I will delete that one and update this one

**Comment by ecoist-ste** [comment]
> Looking great @ecoist-ste ! Could we address [this](https://github.com/gtiosclub/CanvasPlusPlayground/pull/7#discussion_r1759930517)? Or is it not possible?

Sorry I forgot about that one. Let me update this.

File: CanvasPlusPlayground/Views/CourseGradeView.swift (hunk 1 above)
Comment (ecoist-ste) [reply CanvasPlusPlayground/Views/CourseGradeView.swift:52]: On it

**Comment by ecoist-ste** [comment]
> > Looking great @ecoist-ste ! Could we address [this](https://github.com/gtiosclub/CanvasPlusPlayground/pull/7#discussion_r1759930517)? Or is it not possible?
> 
> Sorry I forgot about that one. Let me update this.

@azooz2003-bit Can you check on it again?

File: CanvasPlusPlayground/Views/CourseGradeView.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                if let idLHS = self.course.id, let idRHS = enroll.courseID, idLHS == idRHS  {
+                        self.enrollment = enroll
+                }
+            }
+        }
+    }
+    
+    func display(_ label: String, value: Any?) -> Text {
+        if let value = value {
+            return Text("\(label): \(value)")
```
Comment (rahulon12) [inline CanvasPlusPlayground/Views/CourseGradeView.swift:37]: nit: can make this one line

```swift
Text("\(label): \(value ?? "unavailable")")
```

**Review by rahulon12 — APPROVED** [review]
would like @azooz2003-bit to do a quick scan too

**Comment by azooz2003-bit** [comment]
@ecoist-ste Would like to merge this before Friday. Could you resolve above comments? i.e. We don't need a grade manager class anymore because `CourseManager` can just handle the enrollment fetching. Lmk if there's any confusion.

**Pull request merged**